### PR TITLE
tests: stop using sockets in /tmp for jormungandr_integration_tests

### PR DESCRIPTION
**Pull request by francois-vincent**
http://jira.canaltp.fr/browse/NAVITIAII-2166

### CODE REVIEW

**Comment by TeXitoi** [comment]
Blocked in jormungandr integration tests...

**Comment by francois-vincent** [comment]
@TeXitoi please show me this on next monday !

**Comment by TeXitoi** [comment]
I've just canceled jenkins because it was blocked in jormungandr integration test. I don't know more than that.

File: source/tests/mock_kraken.h [hunk 1]
```diff
@@ -43,6 +45,16 @@ www.navitia.io
 
 namespace po = boost::program_options;
 
+char * workspace() {
```
Comment (TeXitoi) [inline source/tests/mock_kraken.h:48]:
``` c++
static const char* workspace() {
```

File: source/tests/mock_kraken.h [hunk 2]
```diff
@@ -43,6 +45,16 @@ www.navitia.io
...
+    // WORKSPACE is defined on jenkins platform
+    char * ws = getenv("WORKSPACE");
```
Comment (TeXitoi) [inline source/tests/mock_kraken.h:50]: `const char*`

File: source/tests/mock_kraken.h [hunk 3]
```diff
@@ -43,6 +45,16 @@ www.navitia.io
...
+    if (!ws) {
+        // otherwise, use HOME
+        ws = getenv("HOME");
```
Comment (TeXitoi) [inline source/tests/mock_kraken.h:53]: We'll crash badly if HOME is not defined

File: source/tests/mock_kraken.h [hunk 4]
```diff
@@ -43,6 +45,16 @@ www.navitia.io
...
+    };
```
Comment (TeXitoi) [inline source/tests/mock_kraken.h:54]: useless `;`

File: source/jormungandr/tests/tests_mechanism.py [hunk 5]
```diff
@@ -100,9 +100,9 @@ def kill_all_krakens(cls):
     def create_dummy_json(cls):
         conf_template_str = ('{{ \n'
                              '    "key": "{instance_name}",\n'
-                             '    "zmq_socket": "ipc:///tmp/{instance_name}",\n'
+                             '    "zmq_socket": "ipc://%s/{instance_name}",\n'
                              '    "realtime_proxies": {proxy_conf}\n'
-                             '}}')
+                             '}}' % (os.environ.get('WORKSPACE') or os.environ.get('HOME')))
```
Comment (TeXitoi) [inline source/jormungandr/tests/tests_mechanism.py:105]: I'd prefer `/tmp` to directly in my home.

I think something like `(os.environ.get('KRAKEN_BUILD_DIR') or '/tmp')` make more sense as when lanched by nose, `KRAKEN_BUILD_DIR` must be defined to launch the mocks.

**Comment by antoine-de** [comment]
@francois-vincent can you update the pull request ?

**Pull request closed without merging**